var/spack/repos/builtin/packages/faodel/package.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Faodel(CMakePackage):
+    """Flexible, Asynchronous, Object Data-Exchange Libraries"""
+
+    homepage = "https://github.com/faodel/faodel"
+    url      = "https://github.com/faodel/faodel/archive/v1.1811.1.tar.gz"
+    git      = "https://github.com/faodel/faodel.git"
+
+    maintainers = ['fbudin69500', 'chuckatkins']
+
+    version('develop', branch='master')
+    version('1.1811.1', sha256='8e95ee99b8c136ff687eb07a2481ee04560cb1526408eb22ab56cd9c60206916')
+    version('1.1803.1', sha256='70ce7125c02601e14abe5985243d67adf677ed9e7a4dd6d3eaef8a97cf281a16')
+
+    variant('shared', default=False,
+            description='Build Faodel as shared libs')
+    variant('mpi', default=True,
+            description='Enable MPI')
+    variant('hdf5', default=False, description="Build the HDF5-based IOM in Kelpie")
+
+    variant('tcmalloc', default=True,
+            description='Use tcmalloc from gperftools in Lunasa, \
+                         potentially other places')
+    variant('logging', default='stdout', values=('stdout', 'sbl', 'disabled'),
+            description='Select where logging interface output is routed')
+    variant('network', default='nnti', values=('nnti', 'libfabric'),
+            description='RDMA Network library to use for \
+                         low-level communication')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('boost@1.58.0:')
+    depends_on('cmake@3.8.0:')
+    depends_on('hdf5+mpi', when='+hdf5+mpi')
+    depends_on('hdf5~mpi', when='+hdf5~mpi')
+    depends_on('libfabric@1.5.3:', when='network=libfabric')
+
+    # FAODEL v1.1803.1 requires googletest even if tests are disabled
+    depends_on('googletest@1.7.0:', when='@1.1803.1')
```

Comment:
This should be a build time only dependency, i.e. `type='build'`

Reply:
googletest has installed libraries that could be shared.  Doesn't that mean it should be `type=('build', 'link')`?

Reply:
Since it's only for the tests which aren't installed then no.  `type='link'` is to express a private runtime dependency and will ensure it gets added to the rpath of a given package, but not placed into the PATH.  This is a test-only dependency so it's not needed at runtime at all by downstream.